### DISCUSSION TITLE
Enhancing Workflows in Projects

### DISCUSSION THREAD

**Discussion by Jak2k** [community/community#53973] [Projects and Issues] [answered]
### Select Topic Area

Product Feedback

### Select Feature Area

- [ ] Issues
- [X] Projects

### Body

Currently, the automation solution in "Projects" on Github called "Workflows" has some limitations. The workflows provided by Github are helpful, but they are not customizable enough to meet the needs of every project. Therefore, it would be great if Github could provide the option for users to create their own workflows with customizable triggers and effects

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by BenjaminYde** [comment]
This is really needed. There are so many limitations

**Comment by Jak2k** [comment]
Another custom workflow which should be possible: When there is a branch created for an issue, set the custom field "Start Date" to current date.

**Reply by jadshaker** [reply]
Hello @Jak2k , did you find a solution for this?

**Comment by kevinrenskers** [comment]
The existing workflows are really limited. What if you want to change a custom field's value whenever the status is changed? For example: we have a "Progress" select field that we want to reset to the first option every time a card is moved between columns (so, whenever the status is changed). We can't automate this currently.

**Comment by jonagh** [comment]
Another vote for improved workflows for github projects, either 'built in' or via project workflow triggers.

As an example - I'd like to have some kind of basic time tracking for project tasks...
`on (project task fields[status] change) where (fields[status] changed to 'In Progress') then (update fields[started] = now)`
and similar for `...where (fields[status] changed to 'Done') then (update fields[finished] = now)` or something like that.

There is a roadmap item that kinda sounds related to the requested feature:
https://github.com/orgs/github/projects/4247/views/2?filterQuery=projects&pane=issue&itemId=306765

And, although not specifically related to workflows, this roadmap item automates change tracking, and the underlying change triggers could probably be exposed as workflow triggers:
https://github.com/orgs/github/projects/4247/views/2?filterQuery=projects&pane=issue&itemId=37874597

The current suggested workflow automation seems to be based on pull_requests rather than any kind of project change triggers:
https://docs.github.com/en/issues/planning-and-tracking-with-projects/automating-your-project/automating-projects-using-actions

**Reply by probitcarwyn** [reply]
Direct link to the issue:

- https://github.com/github/roadmap/issues/281

**Comment by jubeira** [comment]
+1 to this request.

It'd be great if users could do the following:
- Create new custom workflow.
- Choose workflow trigger from a dropdown with options, which enables or shows a panel to configure the trigger.
- Choose the expected effect from a second dropdown, which also enables or shows a panel to configure the effect.

**Comment by jadshaker** [comment]
Hello,

Did any of you find a solution?

**Comment by jeherve** [comment]
Related discussions:

- https://github.com/orgs/community/discussions/45653
- https://github.com/orgs/community/discussions/53705

**Comment by JosiahParry** [comment]
Currently I want to be able to add a label based on the repository. It would be nice if there was a workflow.
Something like a case when for repository  = {some repo} set label to {some choice}

**Comment by theunhackable** [comment, marked as answer]
Woahh looking at this thread made me realize that the github projects is ded 💀

**Reply by Jak2k** [reply]
Yeah, GitHub as a whole is quite dead. The only stuff they are working on is "AI".

**Comment by probitcarwyn** [comment]
Arguably this has gone down this direction instead: [GitHub Agentic Workflows](https://github.github.com/gh-aw/)

**Comment by linde-ibis** [comment]
Honestly.... why are these so limited? Why can we only have 15? It makes it so un-user friendly for non-techy project managers. We have multiple repos feeding into one project and I can do some things in the YAML issue templates, but not with project fields. (and whyyyy are these YAML and whyyyyyy on a repo-level??? surely there should just be a user interface for these..)
